### DISCUSSION TITLE
Support for mutations without recreating the whole graph?

### DISCUSSION THREAD

**Discussion by serg06** [vasturiano/react-force-graph#555] [Q&A]
Hello! This project has been super useful for rendering static graphs. Now we'd like to use it to render some dynamic graphs as well. E.g. when a user clicks on a node or an edge, we want to delete it. Unfortunately this causes the whole graph to reset and all of the nodes to reposition.

Is there any way to perform this mutation without recreating the graph?

FWIW d3 seems to support it: https://observablehq.com/@d3/modifying-a-force-directed-graph

**Comment by vasturiano** [comment]
@serg06 thanks for reaching out.

Yes, that's easily doable. Please check the [dynamic data changes example](https://vasturiano.github.io/react-force-graph/example/dynamic/). Code is [here](https://github.com/vasturiano/react-force-graph/blob/master/example/dynamic/index.html).

All you have to do is not recreate (or even clone) the objects that are meant to persist from one iteration to the next. The module knows how to recognise if a given object reference was already present so it won't be reset in a data change. It just needs to be the exact same object (by reference).

**Reply by serg06** [reply]
That's perfect thank you!

I was CTRL+F'ing for "mutate", "update", and "add", whereas I should've just looked through the excellent example list.
